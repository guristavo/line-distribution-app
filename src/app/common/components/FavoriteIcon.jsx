import React from 'react';
import PropTypes from 'prop-types';

// Import shared components
import { Icon } from '../../common';

const FavoriteIcon = ({ action, className, id, size, state }) => (
  <button
    className={`btn btn-transparent test ${className}`}
    onClick={() => action(id)}
  >
    {state ? (
      <Icon type="heart" color="red" size={size} />
    ) : (
      <Icon type="heart-hollow" color="gray" size={size} />
    )}
  </button>
);

FavoriteIcon.propTypes = {
  action: PropTypes.func.isRequired,
  className: PropTypes.string,
  id: PropTypes.string.isRequired,
  size: PropTypes.string,
  state: PropTypes.bool,
};

FavoriteIcon.defaultProps = {
  className: '',
  size: 12,
  state: false,
};

export default FavoriteIcon;
