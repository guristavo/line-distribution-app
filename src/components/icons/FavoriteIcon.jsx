import React from 'react';
import PropTypes from 'prop-types';

import favoriteOn from '../../images/icon-favorite.svg';
import favoriteOff from '../../images/icon-favorite-off.svg';

const FavoriteIcon = ({ props }) => {
  const { id } = props.artists.selectedUnit;

  let icon = favoriteOff;
  const isFavorite = props.artists.userFavoriteArtists.findIndex(
    el => el.id === id
  );
  if (isFavorite !== -1) {
    icon = favoriteOn;
  }

  return (
    <button
      className="icon-favorite"
      onClick={() => props.updateFavoriteUnits(id)}
    >
      <img className="icon-favorite-svg" src={icon} alt="Favorite Unit" />
    </button>
  );
};

FavoriteIcon.propTypes = {
  props: PropTypes.object.isRequired,
  artists: PropTypes.object.isRequired,
  updateFavoriteUnits: PropTypes.func.isRequired,
};

export default FavoriteIcon;
