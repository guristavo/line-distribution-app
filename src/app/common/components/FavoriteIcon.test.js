import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import FavoriteIcon from './FavoriteIcon';

describe('common/FavoriteIcon', () => {
  const ID = 'abc';
  const NOOP = () => {};
  let SPY;

  beforeAll(() => {});

  beforeEach(() => {
    SPY = jest.fn();
  });

  afterEach(cleanup);

  it('renders', () => {
    const { asFragment } = render(<FavoriteIcon action={NOOP} id={ID} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('fires action prop function when clicked', () => {
    const { getByTestId } = render(<FavoriteIcon action={SPY} id={ID} />);

    expect(SPY).not.toHaveBeenCalled();

    fireEvent.click(getByTestId('favorite-icon'));

    expect(SPY).toHaveBeenCalledWith(ID);
  });

  it('changes the icon based on the state argument', () => {
    const { getByTitle } = render(<FavoriteIcon action={SPY} id={ID} />);

    expect(getByTitle('heart-hollow')).toBeInTheDocument();

    const { getByTitle: getByTitleWithState } = render(
      <FavoriteIcon action={SPY} id={ID} state />
    );

    expect(getByTitleWithState('heart')).toBeInTheDocument();
  });

  it('accepts custom classes with the className argument', () => {
    const { container } = render(
      <FavoriteIcon action={SPY} id={ID} className="custom-class" />
    );

    expect(container.firstChild.classList.value.trim()).toBe(
      'btn btn--transparent custom-class'
    );
  });
});
