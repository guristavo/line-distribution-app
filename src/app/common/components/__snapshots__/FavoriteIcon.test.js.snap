// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/FavoriteIcon renders 1`] = `
<DocumentFragment>
  <button
    class="btn btn--transparent "
    data-testid="favorite-icon"
  >
    <svg
      height="12px"
      style="display: inline-block; vertical-align: middle; margin: 0px; transform: translateY(-1px);"
      title="heart-hollow"
      viewBox="0 0 100 100"
      width="12px"
    >
      <path
        class="icon-heart-hollow"
        d="M73.2,8c6.4,0,11,1.5,14.1,4.5C90.5,15.6,92,20.1,92,26c0,5.9-3.5,12.4-10.3,19.5L50,75.9L18.3,45.2 c-0.2-0.2-0.4-0.4-0.6-0.5c-0.3-0.3-0.7-0.7-1-1c-0.6-0.8-1.5-1.7-2.5-3c-1-1.2-1.9-2.6-2.9-4.2c-0.1-0.1-0.1-0.2-0.2-0.3 c-0.5-0.8-1.3-2.3-2.2-5c0,0,0-0.1,0-0.1C8.3,29.4,8,27.7,8,26c0-5.9,1.5-10.3,4.7-13.5c3-3,7.6-4.5,14.1-4.5 c1.4,0,2.9,0.3,4.6,0.8c1.7,0.6,3.3,1.3,4.6,2.2c0,0,0.1,0,0.1,0.1c1.5,1,3,2.1,4.4,3.3c0.1,0.1,0.2,0.2,0.4,0.3 c1.3,0.9,2.4,1.9,3.4,3c1.5,1.5,3.5,2.3,5.7,2.3s4.2-0.8,5.7-2.3c1-1,2.2-2,3.4-3c0.1-0.1,0.2-0.2,0.4-0.3c1.4-1.2,2.9-2.3,4.4-3.3 c0,0,0.1,0,0.1-0.1c1.3-0.9,2.9-1.6,4.6-2.2C70.3,8.3,71.9,8,73.2,8 M73.2,0c-2.2,0-4.6,0.4-7,1.2C63.7,2,61.5,3,59.6,4.3 c-1.8,1.2-3.6,2.5-5.3,3.9c-1.6,1.2-3,2.4-4.3,3.7c-1.3-1.3-2.7-2.5-4.3-3.7c-1.7-1.4-3.5-2.7-5.3-3.9c-2-1.3-4.2-2.3-6.6-3.1 C31.3,0.4,29,0,26.8,0C18.2,0,11.6,2.3,7,6.8C2.3,11.5,0,17.9,0,26c0,2.6,0.5,5.2,1.4,7.8c0.9,2.7,1.9,4.9,2.9,6.6 c1.2,2.1,2.4,3.9,3.7,5.5c1.4,1.7,2.5,2.9,3.1,3.7c0.3,0.1,0.6,0.4,0.9,0.8c0.3,0.4,0.6,0.6,0.7,0.6l34.8,33.6c0.7,0.7,1.5,1,2.5,1 s1.9-0.3,2.5-1l34.8-33.4C95.8,42.4,100,34,100,26c0-8.1-2.3-14.5-7-19.1C88.4,2.3,81.8,0,73.2,0L73.2,0z"
        data-testid="icon-path"
        style="fill: #707070;"
      >
        <title>
          heart-hollow
        </title>
      </path>
    </svg>
  </button>
</DocumentFragment>
`;
